components/webapp-mgt/org.wso2.carbon.webapp.mgt/src/main/java/org/wso2/carbon/webapp/mgt/TenantLazyLoaderValve.java

```diff
@@ -94,6 +99,9 @@ public void invoke(Request request, Response response, CompositeValve compositeV
                     } else {
                         request.getRequestDispatcher(requestURI).forward(request, response);
                     }
+                    // Setting the application name to carbon context because it doesn't set the application
+                    // name in re-deployment
+                    carbonContext.setApplicationName(applicationName);
```

Comment:
We need to verify whether this is the right place for this check. Verify the both 'if' and 'else' clauses above.